src/ray/gcs/store_client/redis_store_client.cc

```diff
@@ -164,6 +164,8 @@ Status RedisStoreClient::AsyncGet(const std::string &table_name,
     if (!reply->IsNil()) {
       result = reply->ReadAsString();
     }
+    RAY_CHECK(!reply->IsError())
```

Comment:
After reading the code, I think the right way to do is to update the callback right?

Reply:
Should `callback(Status::OK(), std::move(result));`  become something else?

Reply:
If the difference is between putting it in the callback in KVStore vs the callback in RedisStoreClient, I don't think it matters that much -- the lower callback starts first so crash will happen earlier without bubbling up.

Do you think we need to pass the status back up if we have a RAY_CHECK before it? It's guaranteed to be no error.

Reply:
Synced offline. We can pass it back to the client instead of crashing GCS